src/AzureAuth/Commands/Ado/Pat/CommandScopes.cs

```diff
@@ -0,0 +1,46 @@
+
+using McMaster.Extensions.CommandLineUtils;
+
+using Microsoft.Authentication.AdoPat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Office.Lasso.Telemetry;
+
+using System.Linq;
+
+namespace Microsoft.Authentication.AzureAuth.Commands.Ado.Pat
+{
+    /// <summary>
+    /// Command to print the list of avaialable scopes
+    /// </summary>
+    [Command("scopes", Description = "List the valid ado pat scopes")]
+    public class CommandScopes
+    {
+        private readonly ILogger logger;
+
+        /// <summary>
+        /// Create a CommandScopes
+        /// </summary>
+        /// <param name="logger"></param>
+        public CommandScopes(ILogger<CommandScopes> logger)
+        {
+            this.logger = logger;
+        }
+
+        /// <summary>
+        /// Executes the Scopes command listing the available scopes
+        /// </summary>
+        /// <returns></returns>
+        public int OnExecute()
+        {
+            var scopes = Scopes.ValidScopes.ToList();
+            scopes.Sort();

```

Comment:
Is it possible to have duplicate scopes in the list and so instead we use `HashSet` to only print unique scopes?

Reply:
I believe the motivation for using a HashSet is for constant time lookup for the validation of a set of scopes when running the `ado pat` command. Since HashSet is not inherently ordered, I want to print these alphabetically sorted for ease of use and readability.

Reply:
I guess duplication will never happen as we are just printing from here - https://github.com/AzureAD/microsoft-authentication-cli/blob/6c80faf99de594f0e9958ccd0276fa2c44c2cf45/src/AdoPat/Scopes.cs#L16 :)

Reply:
Their order in `Scopes.ValidScopes` (as defined) is alphabetically by the _groups_ you see in [https://aka.ms/azure-devops-pat-scopes](https://aka.ms/azure-devops-pat-scopes), but we couldn't do that here without changing the data structure that's used.